inmem_endpoint.go

```diff
@@ -41,6 +41,16 @@ type SampledValue struct {
 	DisplayLabels map[string]string `json:"Labels"`
 }
 
+// deepCopy allocates a new instance of AggregateSample
+func (source SampledValue) deepCopy() SampledValue {
+	dest := source
```

Comment:
This actually copies the struct at least two times times, e.g. one in the method call since receiver isn't a reference, and once for the assignment.  I would suggest doing something like:

```go
func (s *SampledValue) deepCopy() SampledValue {
	dest := *s
```

Reply:
thanks, that's better!